proxy/src/telemetry/metrics/labels.rs

```diff
@@ -376,8 +378,14 @@ impl fmt::Display for TlsConfigLabels {
         match self {
             TlsConfigLabels::Reloaded =>
                 f.pad("status=\"reloaded\""),
-            TlsConfigLabels::Io(ref path) =>
-                write!(f, "status=\"io error\",path=\"{}\"", path.display()),
+            TlsConfigLabels::Io { ref path, error_code } => {
+                write!(f, "status=\"io error\",path=\"{}\"", path.display())?;
+                if let Some(error_code) = error_code {
+                    write!(f, ",error_code=\"{}\"", error_code)?;
```

Comment:
After this change, when an IO error is recorded, the counter now also has an `error_code="..."` label with the OS error code for the IO error, if there was one. This seemed like the best way to add information about the kind of IO error without adding the names/`fmt::Display` strings of Rust-specific `io::ErrorKind`s to the labels. However, if anyone has suggestions for some better way to propagate this information, I'd be interested to hear them!